### PR TITLE
Add IntentModal component + use it for IntentOpener :sparkles:

### PR DESCRIPTION
**Pull request by CPatchane**
Be able to invoke an intent modal without using a button/link/node (from a function or whatever, the state will be handled by the application itself).

To keep things coherent, the IntentModal is from the IntentOpener code, now it's split and the IntentModal is used directly by the IntentOpener

### CODE REVIEW

File: react/IntentOpener/IntentOpener.jsx [hunk 1]
```diff
@@ -53,21 +52,16 @@ class IntentOpener extends React.Component {
...
-          <IntentIframe
-            action={action}
-            doctype={doctype}
-            options={options}
-            onComplete={this.handleComplete}
-            create={create}
-          />
-        </Modal>
+        <IntentModal
+          closable={closable}
```
Comment (ptbrowne) [inline react/IntentOpener/IntentOpener.jsx]: The key has been removed, is it on purpose ?

File: react/IntentModal/IntentModal.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+IntentModal.propTypes = {
+  /** What should happen when the intent has completed */
+  onComplete: PropTypes.func.isRequired,
+  /** What to do to dismiss the modal */
+  dismissAction: PropTypes.func.isRequired,
+  /** Action you want to execute */
+  action: PropTypes.string.isRequired,
+  /** Doctype on which you want to execute the action */
+  doctype: PropTypes.string.isRequired
```
Comment (ptbrowne) [inline react/IntentModal/IntentModal.jsx]: Is there a reason why closable is not documented ?

File: react/IntentModal/IntentModal.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+```
```
Comment (ptbrowne) [inline react/IntentModal/IntentModal.md]: Can you add a text explaining what's a usecase for this ?

Comment (CPatchane) [reply react/IntentModal/IntentModal.md]: Sure!

File: react/IntentModal/IntentModal.jsx (hunk 2 above)
Comment (CPatchane) [reply react/IntentModal/IntentModal.jsx]: No, I added it so 👍

**Comment by CPatchane** [comment]
[(2) ETU j'ai la modale d'inter-app de collect à la fin de l'installation d'un connecteur](https://trello.com/c/470Nuja1/895-2-etu-jai-la-modale-dinter-app-de-collect-%C3%A0-la-fin-de-linstallation-dun-connecteur)

**Pull request merged**